src/tir/analysis/deep_equal.cc

```diff
@@ -59,6 +59,9 @@ bool ExprDeepEqual::operator()(const PrimExpr& lhs, const PrimExpr& rhs) const {
     auto* prhs = rhs.as<IntImmNode>();
     return plhs->dtype == prhs->dtype && plhs->value == prhs->value;
   }
+  if (lhs.as<AnyNode>()) {
+    return lhs.same_as(rhs);
```

Comment:
```suggestion
    return false;
```
the code that check same_as already happend at before.

Reply:
Nice catch!